api/internal/localizer/localizer.go

```diff
@@ -0,0 +1,17 @@
+// Copyright 2022 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package localizer
+
+import (
+	"log"
+
+	"sigs.k8s.io/kustomize/kyaml/errors"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+func cleanDst(newDir string, fSys filesys.FileSystem) {
+	if err := fSys.RemoveAll(newDir); err != nil {
+		log.Printf("%s", errors.WrapPrefixf(err, "unable to clean localize destination").Error())
```

Comment:
In general, we should return errors wherever possible instead of logging. It will be easier to reason about what went wrong (and to debug).

For example, if a calls b and b calls c and c logs an error, we will have to manually figure out what was the stack trace ourselves. But if instead we return errors along the way, we will first discover that a throws an error because b throws an error because c throws an error - it is easier to reason about and also allows us to have more descriptive errors.

Besides all of that, it is usually the Golang way for functions/methods to return an error as a last return value whenever an error is possible.

Reply:
Agreed. I'd also like to return this error. However, I only call this function when some error is thrown in the calling function, and so if I run into this error, I'd have to return both errors or wrap them. Returning both errors seems clunky and wrapping them seemed strange because they technically aren't the same error. I logged the error because I was at a loss for what to do.

Reply:
IMO when you add a function, you can't assume that you will be the only caller; the function should be written in a way that makes sense for all callers. In most cases, it doesn't make sense to log an error without returning it. With that in mind, I still think that (a) if we do have this function, then we should return an error and (b) if we do that, then it doesn't make it any different from running `fSys.RemoveAll(newDir)`.

Re: having to return both errors: in such a case we can either ignore or log the error returned by `fSys.RemoveAll` at the point that it is used. This function still seems like an overcomplication.

Reply:
Sure, that sounds good. I'll move the logging into the calling function.